packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -233,7 +233,7 @@ export class ProjectLifecycleManager {
           const devServerOptions = await this.ctx._apis.projectApi.getDevServer().start({ specs: this.ctx.project.specs, config: finalConfig })
 
           if (!devServerOptions?.port) {
-            this.ctx.onError(getError('CONFIG_FILE_DEV_SERVER_INVALID_RETURN', devServerOptions))
+            throw getError('CONFIG_FILE_DEV_SERVER_INVALID_RETURN', devServerOptions)
```

Comment:
Should this be named something else? If it is the return type from the dev server function, should it be something like `devServerReturnVal`?

Reply:
I think this object is what we get from here: https://github.com/cypress-io/cypress/blob/develop/npm/webpack-dev-server/src/devServer.ts#L62-L76